test/matchers/to-have-status-effect-matcher.ts

```diff
@@ -0,0 +1,53 @@
+// -- start tsdoc imports --
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { Pokemon } from "#app/field/pokemon";
+// -- end tsdoc imports --
+
+import { StatusEffect } from "#enums/status-effect";
+import { isPokemonInstance, receivedStr } from "#test/test-utils/testUtils";
+import type { SyncExpectationResult } from "@vitest/expect";
```

Comment:
Add `MatcherState` to be able to add type-hints for `this`

```suggestion
import type { MatcherState, SyncExpectationResult } from "@vitest/expect";
```